src/lib.rs

```
@@ -142,6 +145,49 @@ impl Cookie {
     }
 }
 
+#[cfg(feature = "serialize-serde")]
+impl Serialize for Cookie {
+    fn serialize<S>(&self, serializer: &mut S) -> Result<(), S::Error>
+        where S: serde::Serializer
+    {
+        serializer.serialize_str(&*self.to_string())
+    }
+}
+
+#[cfg(feature = "serialize-serde")]
+impl Deserialize for Cookie {
+    fn deserialize<D>(deserializer: &mut D) -> Result<Cookie, D::Error>
+        where D: serde::Deserializer
+    {
+        deserializer.deserialize_string(CookieVisitor)
+    }
+}
+#[cfg(feature = "serialize-serde")]
+struct CookieVisitor;
+
+#[cfg(feature = "serialize-serde")]
+impl serde::de::Visitor for CookieVisitor {
+    type Value = Cookie;
+
+    fn visit_str<E>(&mut self, v: &str) -> Result<Cookie, E>
+        where E: serde::de::Error
+    {
+        match Cookie::parse(v) {
+            Ok(cookie) => Ok(cookie),
+            Err(_) => Err(serde::de::Error::custom("Could not parse serialized cookie!"))
```

Could this pass along the error instead of swallowing it up?